src/index.ts

```diff
@@ -316,6 +300,7 @@ const plugin: JupyterFrontEndPlugin<void> = {
           manuallySharing.add(notebookPanel);
 
           // Save the notebook before we share it.
+          manualSaveCount++;
           await notebookPanel.context.save();
 
           await handleNotebookSharing(notebookPanel, sharingService, true);
```

Comment:
- Why not put it in `handleNotebookSharing` to avoid duplication and ensure it is always set?
- Why use counter over boolean?

Reply:
Good points. I'll rename the counter to a boolean like `hasManuallySaved`, as the UI only needs to know whether a manual save has ever occurred, not how many times.

Reply:
Addressed in 4f616399a337f4ace63881c5f574af019b319e32, please let me know. Thanks!